src/components/Infra/InfraEnvForm.tsx

```diff
@@ -37,18 +42,19 @@ const InfraEnvWizard: React.FC<CreateResourceComponentProps> = ({ namespace = 'd
     async (values) => {
       // TODO(mlibra): Secret, clusterDeployment and agentClusterinstall should be removed from here once we have Late Binding
       // TODO(mlibra): We already have, let's do it!
-      const secret = await k8sCreate({
+      const secret = await k8sCreate<SecretK8sResource>({
         model: secretModel,
         data: getSecret(namespace, values),
       });
-      const clusterDeployment = await k8sCreate({
+      const clusterDeployment = await k8sCreate<ClusterDeploymentK8sResource>({
         model: clusterDepModel,
-        data: getClusterDeploymentForInfraEnv(secret.metadata.name, namespace, values),
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        data: getClusterDeploymentForInfraEnv(secret.metadata?.name!, namespace, values),
```

Comment:
It is unfortunate that `K8sResourceCommon` defines all metadata as optional. It would be good to enhance it to reflect the reality and make the required metadata properties such as name and namespace required.